lib/src/Uncertainty/Bayesian/RandomWalkMetropolisHastings.cxx

```diff
@@ -122,11 +125,8 @@ void RandomWalkMetropolisHastings::setProposal(const Distribution & proposal)
   if (proposal.getDimension() != marginalIndices_.getSize())
     throw InvalidArgumentException(HERE) << "The proposal density dimension (" << proposal.getDimension()
                                          << ") does not match the block size (" << marginalIndices_.getSize() << ")";
-  if (!(proposal.getSkewness().norm() < ResourceMap::GetAsScalar("Distribution-DefaultQuantileEpsilon")))
-    throw InvalidArgumentException(HERE) << "The proposal density is not symmetric.";
-  if (!(proposal.getMean().norm() < ResourceMap::GetAsScalar("Distribution-DefaultQuantileEpsilon")))
-    throw InvalidArgumentException(HERE) << "The proposal density must have a null mean.";
   proposal_ = proposal;
+  isProposalSymmetric_ = proposal.isElliptical() && (proposal.getMean().norm() < ResourceMap::GetAsScalar("Distribution-DefaultQuantileEpsilon"));
```

Comment:
This test is correct in the 1D case, where symmetry is equivalent to ellipticity, but what is the meaning of symmetry in the multivariate case? Is it equivalent to ellipticity? Or to independence between symmetrical marginals?

Reply:
Symmetry means that for any vector t, the PDF at (-t) is equal to the PDF at t.

(Ellipticity + zero mean) is a sufficient condition for symmetry regardless of the dimension, but I do not think it is necessary when the dimension is >1. For example the uniform distribution on [-1,1] x [-1, 1] is symmetric, but I do not think it is elliptical.

Independence between symmetric marginals is another sufficient condition for symmetry, indeed, but it is also not necessary. It could however be added, as it covers many situations in practice.

I am unaware of any easily verifiable sufficient and necessary criterion for symmetry, so I picked a sufficient one. Sampling will remain correct if we miss symmetry, but will involve unnecessary computations.

An alternative idea could be to monitor the values of `probCurrentConditionedToNew_` and `probNewConditionedToCurrent_` in the first iterations of the algorithm, and to switch `isProposalSymmetric_` to true if they are always found to be equal. This might be more trouble than it is worth, though.

Reply:
Ok so IMO at least a comment should be added in the code to explain that. Otherwise it is ok for me.

Reply:
Both conditions are now tested. Comments in the code summarize this discussion.